### PR TITLE
Alerting: add field for custom slack endpoint

### LINKED ISSUE #45119: Slack notification channel: support alternative API URL for Slack
### Discussed in https://github.com/grafana/grafana/discussions/44876

We should add a new field that overrides the domain name in `slackAPIEndpoint` in each notification channel, this is required for users of https://slack-gov.com.

### PR DESCRIPTION
**Pull request by nathanrodman**
**What this PR does / why we need it**:

Adds a field for adding a custom slack url. This is useful for users that have a different domain other than `slack.com` such as government users.

**Which issue(s) this PR fixes**:

Fixes #45119 

**Special notes for your reviewer**:

I need to update the tests but could use some help in figuring out the best way to test the settings.

### CODE REVIEW

File: pkg/services/ngalert/notifier/channels/slack.go [hunk 1]
```diff
@@ -52,17 +52,22 @@ func NewSlackNotifier(model *NotificationChannelConfig, t *template.Template, fn
 		return nil, receiverInitError{Cfg: *model, Reason: "no secure settings supplied"}
 	}
 
+	endpointURL := model.Settings.Get("endpointUrl").MustString()
+	if endpointURL == "" {
+		endpointURL = SlackAPIEndpoint
+	}
```
Comment (alexweav) [inline pkg/services/ngalert/notifier/channels/slack.go:58]:
```suggestion
	endpointURL := model.Settings.Get("endpointUrl").MustString(SlackAPIEndpoint)
```
`MustString` takes a variadic arg, which is the default to use if the key does not exist.

File: pkg/services/ngalert/notifier/available_channels.go [hunk 2]
```diff
@@ -459,6 +459,14 @@ func GetAvailableNotifiers() []*alerting.NotifierPlugin {
 					PropertyName: "url",
 					Secure:       true,
 				},
+				{
+					Label:        "Endpoint URL",
+					Element:      alerting.ElementTypeInput,
+					InputType:    alerting.InputTypeText,
+					Description:  "Optionally provide a custom Slack message API endpoint, default is https://slack.com/api/chat.postMessage",
```
Comment (alexweav) [inline pkg/services/ngalert/notifier/available_channels.go:466]: Is it worth mentioning in the description that this is only for non-webhook requests? The webhook approach would already contain the correct base URL inside the webhook config itself, and this setting would have no impact.

Comment (alexweav) [reply pkg/services/ngalert/notifier/available_channels.go:466]: Many of the other settings do this too - once that are either webhook-specific or not mention this in their descriptions.

**Review by alexweav — COMMENTED** [review]
Looking great - just needs a test. Comments were only very small nit-picks.

**Comment by srclosson** [comment]
Moving this PR to 8.5.0

File: pkg/tests/api/alerting/api_available_channel_test.go [hunk 3]
```diff
@@ -46,1654 +48,9 @@ func TestAvailableChannels(t *testing.T) {
 	b, err := ioutil.ReadAll(resp.Body)
 	require.NoError(t, err)
 	require.Equal(t, 200, resp.StatusCode)
-	require.JSONEq(t, expAvailableChannelJsonOutput, string(b))
-}
 
-var expAvailableChannelJsonOutput = `
```
Comment (alexweav) [inline pkg/tests/api/alerting/api_available_channel_test.go:52]: image

**Review by alexweav — APPROVED** [review]
LGTM

**Pull request merged**